buildSrc/src/main/groovy/org/elasticsearch/gradle/test/DistroTestPlugin.java

```diff
@@ -405,4 +465,80 @@ private static String destructiveDistroTestTaskName(ElasticsearchDistribution di
             distro.getFlavor(),
             distro.getBundledJdk());
     }
+
+    static Map<String, String> parseOsRelease(final List<String> osReleaseLines) {
+        final Map<String, String> values = new HashMap<>();
+
+        for (String line : osReleaseLines) {
+            final String trimmed = line.trim();
+
+            // ignore empty and comment lines
+            if (trimmed.isEmpty() || trimmed.startsWith("#")) {
+                continue;
+            }
+
+            final String[] parts = line.split("=", 2);
+            final String key = parts[0];
+            // remove optional leading and trailing quotes and whitespace
+            final String value = parts[1].replaceAll("^['\"]?\\s*", "").replaceAll("\\s*['\"]?$", "");
+            values.put(key, value);
+        }
+
+        return values;
+    }
+
+    static String deriveId(final Map<String, String> osRelease) {
+        return osRelease.get("ID") + "-" + osRelease.get("VERSION_ID");
+    }
+
+    /**
+     * The {@link DistroTestPlugin} generates a number of test tasks, some
+     * of which are Docker packaging tests. When running on the host OS
+     * i.e. not in a VM, only certain operating systems are supported. This
+     * method determines whether the Docker tests should be run on the host
+     * OS.
+     */
+    private static boolean shouldRunDockerTests() {
+        switch (OS.current()) {
+            case WINDOWS:
+                // Not currently supported.
+                logger.debug("shouldRunDockerTests = false on " + OS.WINDOWS);
+                return false;
+
+            case MAC:
+                // Assume that Docker for Mac is installed
+                logger.debug("shouldRunDockerTests = true on " + OS.MAC);
+                return true;
+
+            case LINUX:
+                final Path osRelease = Paths.get("/etc/os-release");
+
+                if (Files.exists(osRelease)) {
+                    Map<String, String> values;
+
+                    try {
+                        final List<String> osReleaseLines = Files.readAllLines(osRelease);
+                        values = parseOsRelease(osReleaseLines);
+                    } catch (IOException e) {
+                        throw new GradleException("Failed to read /etc/os-release", e);
+                    }
+
+                    final String id = deriveId(values);
+
+                    final boolean contains = DOCKER_LINUX_INCLUDE_LIST.contains(id);
+
+                    logger.debug("Linux OS id [" + id + "] is " + (contains ? "" : "not ") + "present in the include list");
```

Comment:
Here's the thing about debug logging... it's never on in CI, and if you turn it on locally, the amount of shit that gets logged pretty much drowns out anything you might otherwise be looking for. If this is useful for local debugging I'd bump it to `info`, otherwise include in an exception message where appropriate.